### ISSUE TITLE
Theme switching - light option

### ISSUE THREAD

**Issue by bremensaki** [nioc/xmpp-web#38]
A light theme is necessary as dark-only has accessibility problems, especially for users with certain eyesight issues. A switch option for a brighter display needs to be prominent and available before login, or logging in itself might be very difficult for people.

Being able to set a different default server-side would be good too.

**Comment by nioc** [comment]
Hello @bremensaki, accessibility and user experience are 2 different things.

You said "_a light theme is necessary_", I disagree with that. It is an UX choice (mine, I love dark themes 😆). If you want to argue or explain in details, feel free to reply/send a message.

I check accessibility ([color contrast as defined by WCAG guideline](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Color_contrast), to be exact) and most of the app is more than ok (greater than 7, which is enhanced ratio "AAA rating"), for example:
- 9.87 for messages text
- 13.74 for contacts
- 8.82 for form inputs

The following items need to be improved (lower than 4.5, the minimum ratio "AA rating"):
- active chat/room (white on green: 3.17),
- unread messages counter (white on pink: 3.61),
- `is-primary` class button (white on green: 3.17).

Let me know if there is any other items to improve.

**Comment by bremensaki** [comment]
There's a pretty good summary of the issues with dark themes here: https://levelup.gitconnected.com/why-dark-mode-causes-more-accessibility-issues-than-it-solves-d2f8359bb46a

Predominantly dark pages cause issues for many people with forms of astigmatism, myself included, which increases eyestrain massively. It's not a style choice, it's literally painful.

Some sort of switching is the only way you're going to cover this, contrast is only one factor - although definitely a huge and necessary one. I'm happy to do some work on a light theme itself and will probably fork this in the next day or so to play around with it on my servers, but not sure where to start on switching methods.

**Comment by nioc** [comment]
Hello, ok you convinced me. I'm myopic and astigmatic but I get glasses 🤓...

If you have time, it would be great if you can propose a PR for this.

I have already implemented a light/dark mode switch on another project, you can check the commit https://github.com/nioc/vue-dom/commit/ef6a1ab4a8537ae1fead736a61b16c6906e93c2a:
- adapt the CSS [styles.scss#L40-L319](https://github.com/nioc/vue-dom/blob/master/src/assets/styles.scss#L40-L319)
- add a switch element somewhere:
  - [switch Sidebar.vue#L86-L90](https://github.com/nioc/vue-dom/blob/master/src/components/Sidebar.vue#L86-L90),
  - [declare variable Sidebar.vue#L117](https://github.com/nioc/vue-dom/blob/master/src/components/Sidebar.vue#L117),
  - [initialize variable Sidebar.vue#L130-L134](https://github.com/nioc/vue-dom/blob/master/src/components/Sidebar.vue#L130-L134) and
  - [handle switch event Sidebar.vue#L143-L153](https://github.com/nioc/vue-dom/blob/master/src/components/Sidebar.vue#L143-L153).
- read the value in local storage at app starting [App.vue#L56-L60](https://github.com/nioc/vue-dom/blob/master/src/App.vue#L56-L60)

**Comment by bremensaki** [comment]
Sure, I'll see if I get time in the next few days to take a stab. Never done anything with Vue but the reference code looks pretty straightforward, thanks for that.

**Comment by nioc** [comment]
If you want I can do the Vue part (the easy part for me), but the SCSS is a lot of work and I do not have time to handle it for now...

**Comment by bremensaki** [comment]
I'll see how I go with the theme changes first, since that's probably the most sensible first step. If you take any steps along the path of adding a switcher ping me.

**Comment by nioc** [comment]
I'll take a look to the button tonight (in the next 8 hours…).
As it should be added to multiple places (login, guest home, chat), it must be a dedicated component with its own logic (including local storage update).

**Comment by nioc** [comment]
I created a [dedicated branch](https://github.com/nioc/xmpp-web/tree/light-mode) for this feature until CSS is done.

Switch is done which add a `data-theme="light"` to the DOM, save and read to local storage. You have to add the following to SCSS file (at least after line 69):
```scss
[data-theme="light"] {
    /* colors changes here */
}
```

Here are the screenshots:
- login:
![image](https://user-images.githubusercontent.com/5225411/128640796-d80c82b8-30c5-4c39-b8a5-095ad41863fe.png)

- guest:
![image](https://user-images.githubusercontent.com/5225411/128640815-65925e6d-3581-47d4-a380-779b1404fe5c.png)

- chat:
![image](https://user-images.githubusercontent.com/5225411/128640839-97aa855b-b9ce-4678-8779-518d328f3a0e.png)

**Comment by bremensaki** [comment]
OK I get how it all works now, and I've identified most of the key areas I think. Just need to chip away at all the various elements in all the views now and juggle things around until it looks pleasant enough.

I'll use the defined primary as a prominent accent colour, but other than that I'll keep to some simple standards for the first pass, rather than start adding in new definitions all over the place.

![image](https://user-images.githubusercontent.com/4737927/128861006-64a92cf0-1af8-492e-9ec0-2d1178f398ab.png)

**Comment by nioc** [comment]
Close for inactivity
